helix-rest/src/main/java/org/apache/helix/rest/metadatastore/ZkMetadataStoreDirectory.java

```diff
@@ -119,14 +119,20 @@ public ZkMetadataStoreDirectory(Map<String, String> routingZkAddressMap)
 
   @Override
   public Map<String, String> getAllMappingUnderPath(String namespace, String path) {
-    // TODO: get it from routingData
-    throw new UnsupportedOperationException();
+    if (!_routingDataMap.containsKey(namespace)) {
+      throw new IllegalArgumentException(
+          "Failed to get all mapping under path: Namespace " + namespace + " is not found!");
```

Comment:
Nit: let's use `NoSuchElementException` to keep it consistent with other methods in MSD?

Reply:
The other methods in MSD use `IllegalArgumentException` such as the exact same check in `addMetadataStoreRealm`

Reply:
> throw new NoSuchElementException("Namespace " + namespace + " does not exist!"); 
?

For read methods, please use NoSuchElement where appropriate.
For write methods, use IllegalArgument?

Reply:
I see what you mean. Agreed, changing.